crates/apps_lib/src/cli/wallet.rs

```diff
@@ -184,13 +184,14 @@ fn payment_addresses_list(
 }
 
 /// Derives a masp spending key from the mnemonic code in the wallet.
-fn shielded_key_derive(
+async fn shielded_key_derive(
```

Comment:
why did this become async?

Reply:
it's taken from #3797 without shielded keys support in HW wallet - we unfortunately already have some blocking calls mixed in the client already (e.g. wallet load) but I think we should just go with it and convert the blocking stuff to async